app/src/main/java/org/wikipedia/page/PageAvailableOfflineHandler.kt

```diff
@@ -16,19 +18,16 @@ object PageAvailableOfflineHandler {
         callback.onFinish(WikipediaApp.instance.isOnline || (page.offline && !page.saving))
     }
 
-    @SuppressLint("CheckResult")
     fun check(pageTitle: PageTitle, callback: Callback) {
         if (WikipediaApp.instance.isOnline) {
             callback.onFinish(true)
             return
         }
-        CoroutineScope(Dispatchers.Main).launch(CoroutineExceptionHandler { _, exception ->
-            run {
-                callback.onFinish(false)
-                L.w(exception)
-            }
+        CoroutineScope(Dispatchers.IO).launch(CoroutineExceptionHandler { _, exception ->
```

Comment:
Can we find a way to pass a `coroutineScope` into this function from the caller?

Reply:
The `setHistoryEntry` where the `check` function is used, will be called from `Fragment` and `view` class.

It is easy to use `lifecycleScope` from a `Fragment`, but not from a `view` class. I assume that if we want to do that, we will need to update the `ListCardView` and the parent class, not sure if there is another way of calling `lifecycleScope` instead of using `CoroutineScope` 🤔

Reply:
A `View` has a `context`, which is usually the Activity that owns the view, so we should be able to do something like `(view.context as? AppCompatActivity)?.lifecycleScope ?: CoroutineScope(IO)`

Reply:
Thanks! Done.